src/main/java/dev/openfeature/sdk/providers/memory/InMemoryProvider.java

```diff
@@ -0,0 +1,184 @@
+            return ProviderEvaluation.<T>builder()
+                .errorCode(errorCode)
+                .reason(errorCode.name())
+                .value(defaultValue)
+                .build();
+        }
+        Flag<?> flag = flags.get(key);
+        if (flag == null) {
+            return ProviderEvaluation.<T>builder()
+                .value(defaultValue)
+                .reason(Reason.ERROR.toString())
+                .errorMessage(ErrorCode.FLAG_NOT_FOUND.name())
+                .errorCode(ErrorCode.FLAG_NOT_FOUND)
+                .build();
+        }
+        T value;
+        if (flag.getContextEvaluator() != null) {
+            value = (T) flag.getContextEvaluator().evaluate(flag, evaluationContext);
+        } else if (!expectedType.isInstance(flag.getVariants().get(flag.getDefaultVariant()))) {
+            return ProviderEvaluation.<T>builder()
+                .value(defaultValue)
+                .variant(flag.getDefaultVariant())
+                .reason(Reason.ERROR.toString())
+                .errorMessage(ErrorCode.TYPE_MISMATCH.name())
+                .errorCode(ErrorCode.TYPE_MISMATCH)
+                .build();
+        } else {
+            value = (T) flag.getVariants().get(flag.getDefaultVariant());
+        }
+        return ProviderEvaluation.<T>builder()
+            .value(value)
+            .variant(flag.getDefaultVariant())
+            .reason(Reason.STATIC.toString())
+            .build();
+    }
+
+    private static Value objectToValue(Object object) {
+        if (object instanceof Value) {
+            return (Value) object;
+        } else if (object == null) {
+            return null;
+        } else if (object instanceof String) {
+            return new Value((String) object);
+        } else if (object instanceof Boolean) {
+            return new Value((Boolean) object);
+        } else if (object instanceof Integer) {
+            return new Value((Integer) object);
+        } else if (object instanceof Double) {
+            return new Value((Double) object);
+        } else if (object instanceof Structure) {
+            return new Value((Structure) object);
+        } else if (object instanceof List) {
+            return new Value(((List<Object>) object).stream()
+                .map(InMemoryProvider::objectToValue)
+                .collect(Collectors.toList()));
+        } else if (object instanceof Instant) {
+            return new Value((Instant) object);
+        } else if (object instanceof Map) {
+            return new Value(mapToStructure((Map<String, Object>) object));
+        } else {
+            throw new TypeMismatchError("Flag value " + object + " had unexpected type "
+                    + object.getClass() + ".");
+        }
+    }
+
+    /**
+     * transform a map to a Structure type.
+     *
+     * @param map map of objects
+     * @return a Structure object in the SDK format
+     */
+    public static Structure mapToStructure(Map<String, Object> map) {
```

Comment:
This can be exposed from `MutableStructure` class so that it can be commonly used where needed